elephant/signal_processing.py

```diff
@@ -611,3 +611,160 @@ def hilbert(signal, N='nextpow'):
+        Default: None
+    baseline : string or quantities.Quantity
+        A factor to subtract from the signal before rectification. If `'mean'`
+        or `'median'`, the mean or median value of the entire signal is
+        subtracted on a channel-by-channel basis.
+        Default: None
+    t_start, t_stop : quantities.Quantity
+        Times to start and end the algorithm. The signal is cropped using
+        `signal.time_slice(t_start, t_stop)` after baseline removal. Useful if
+        you want the RAUC for a short section of the signal but want the
+        mean or median calculation (`baseline='mean'` or `baseline='median'`)
+        to use the entire signal for better baseline estimation.
+        Default: None
+
+    Returns
+    -------
+    quantities.Quantity or neo.AnalogSignal
+        If the number of bins is 1, the returned object is a scalar or
+        vector Quantity containing a single RAUC value for each channel.
+        Otherwise, the returned object is an AnalogSignal containing the
+        RAUC(s) for each bin stored as a sample, with times corresponding to
+        the center of each bin. The output signal will have the same number
+        of channels as the input signal.
+
+    Raises
+    ------
+    TypeError
+        If the input signal is not a neo.AnalogSignal.
+    TypeError
+        If `bin_duation` is not None or a Quantity.
+    TypeError
+        If `baseline` is not None, `'mean'`, `'median'`, or a Quantity.
+    '''
+
+    if not isinstance(signal, neo.AnalogSignal):
+        raise TypeError('Input signal is not a neo.AnalogSignal!')
+
+    if baseline is None:
+        pass
+    elif baseline is 'mean':
+        # subtract mean from each channel
+        signal = signal - signal.mean(axis=0)
+    elif baseline is 'median':
+        # subtract median from each channel
+        signal = signal - np.median(signal.as_quantity(), axis=0)
+    elif isinstance(baseline, pq.Quantity):
+        # subtract arbitrary baseline
+        signal = signal - baseline
+    else:
+        raise TypeError(
+            'baseline must be None, \'mean\', \'median\', '
+            'or a Quantity: {}'.format(baseline))
+
+    # slice the signal after subtracting baseline
+    signal = signal.time_slice(t_start, t_stop)
+
+    if bin_duration is not None:
+        # from bin duration, determine samples per bin and number of bins
+        if isinstance(bin_duration, pq.Quantity):
+            samples_per_bin = int(np.round(
+                bin_duration.rescale('s')/signal.sampling_period.rescale('s')))
+            n_bins = int(np.ceil(signal.shape[0]/samples_per_bin))
+        else:
+            raise TypeError(
+                'bin_duration must be a Quantity: {}'.format(bin_duration))
+    else:
+        # all samples in one bin
+        samples_per_bin = signal.shape[0]
+        n_bins = 1
+
+    # store the actual bin duration
+    bin_duration = samples_per_bin * signal.sampling_period.rescale('s')
```

Comment:
Here, the time stamps of the output of the rauc function is always in units of seconds, in case a binning is applied. Another option would be to transform the output of the rauc function to the time units of the input AnalogSignal. I think there are arguments for both ways of doing it, personally I favor the latter. What is your take?

Reply:
I agree, the result should have the same time units as the input. I've fixed this in ceb18f8.